backend/prompt_studio/prompt_profile_manager_v2/models.py

```diff
@@ -0,0 +1,115 @@
+import uuid
+
+from account_v2.models import User
+from adapter_processor_v2.models import AdapterInstance
+from django.db import models
+from prompt_studio.prompt_studio_core_v2.exceptions import DefaultProfileError
+from prompt_studio.prompt_studio_core_v2.models import CustomTool
+from utils.models.base_model import BaseModel
+
+
+class ProfileManager(BaseModel):
```

Comment:
Don't we need org info here @muhammad-ali-e ?

Reply:
Profile is tightly coupled with Tools.So Ithink  it won't required. Right?

Reply:
But if we are looking out for multi-tenancy, we will need org id stamped against the major entities that we maintain. Here, I am not very clear on how we maintainthe relationship between tools and orgs. But we will need that relationship. If you say that by using tools of an org, we can pull profiles of the same org, it may be ok for now. But a clean separation is preferrable in my opinion @muhammad-ali-e 
CC: @hari-kuriakose

Reply:
@muhammad-ali-e Adding to this, if we decide to scale any model in any new product calls, adding org_id will need additional migration efforts. It would be better to have org_id in all new models, irrespective of how they are coupled or accessed.

Reply:
@muhammad-ali-e Adding to this, if we decide to scale any model in any new product calls, adding org_id will need additional migration efforts. It would be better to have org_id in all new models, irrespective of how they are coupled or accessed.

Reply:
@muhammad-ali-e Adding to this, if we decide to scale any model in any new product calls, adding org_id will need additional migration efforts. It would be better to have org_id in all new models, irrespective of how they are coupled or accessed.

Reply:
@hari-kuriakose could you please confirm if we need to establish an organization relation for every table, even if it's not immediately necessary? 
 1. I believe it is straightforward to retrieve the organization ID from customTool, which already has this relationship. Currently, we don't need a direct relation with the organization for the ProfileManager model; we only need the customTool relation. If we require this relation in the future, we can create a migration to add the organization ID, which we can obtain from the too.
2. Adding unnecessary relationships will increase complexity and resource utilization, and impact performance. 
3. The only potential benefit is avoiding future migrations if the relationship is needed later. However, I don't think we will need it in the future either.

Reply:
@muhammad-ali-e Agree with you here.

Folks, the nature of RDBMS is not to de-centralize foreign keys, instead to centralize concepts and leverage the power of joins.

That being said, assuming we have a join path from profile to org. Now since the foreign keys in respective models are already indexed the performance should also be ok, should we need this query in the future.

Reply:
I am ok with not adding it for now as long as there is way to cleanly pull entities of an org and give the user , the multi-tenant experience! @muhammad-ali-e @hari-kuriakose